iam-login-service/src/main/java/it/infn/mw/iam/api/scim/provisioning/ScimUserProvisioning.java

```diff
@@ -146,36 +144,43 @@ public ScimUser create(final ScimUser user) {
       IamAccount account = accountService.createAccount(newAccount);
       return userConverter.dtoFromEntity(account);
     } catch (CredentialAlreadyBoundException | UserAlreadyExistsException e) {
-      throw new ScimResourceExistsException(e.getMessage(),e);
+      throw new ScimResourceExistsException(e.getMessage(), e);
     }
   }
 
   @Override
   public ScimListResponse<ScimUser> list(final ScimPageRequest params) {
 
+    ScimListResponseBuilder<ScimUser> builder = ScimListResponse.builder();
+
     if (params.getCount() == 0) {
-      int userCount = accountRepository.countAllUsers();
-      return new ScimListResponse<>(Collections.emptyList(), userCount, 0, 1);
-    }
 
-    OffsetPageable op = new OffsetPageable(params.getStartIndex(), params.getCount());
+      long totalResults = accountRepository.countAllUsers();
+      builder.totalResults(totalResults);
+
+    } else {
 
-    Page<IamAccount> results = accountRepository.findAll(op);
+      OffsetPageable op = new OffsetPageable(params.getStartIndex(), params.getCount());
 
-    List<ScimUser> resources = new ArrayList<>();
+      Page<IamAccount> results = accountRepository.findAll(op);
 
-    results.getContent().forEach(a -> resources.add(userConverter.dtoFromEntity(a)));
+      List<ScimUser> resources = new ArrayList<>();
+
+      results.getContent().forEach(a -> resources.add(userConverter.dtoFromEntity(a)));
+
+      builder.resources(resources);
+      builder.fromPage(results, op);
+    }
 
-    return new ScimListResponse<>(resources, results.getTotalElements(), resources.size(),
-        op.getOffset() + 1);
+    return builder.build();
   }
 
   @Override
   public ScimUser replace(final String uuid, final ScimUser scimItemToBeUpdated) {
 
     // user must exist
-    IamAccount existingAccount = accountRepository.findByUuid(uuid)
-      .orElseThrow(() -> new ScimResourceNotFoundException("No user mapped to id '" + uuid + "'"));
+    IamAccount existingAccount = accountRepository.findByUuid(uuid).orElseThrow(
+        () -> new ScimResourceNotFoundException("No user mapped to id '" + uuid + "'"));
```

Comment:
Use String.format() to build the exception message

Reply:
I would have a method to build this exception since is used in lot of places ( avoid cut & paste when coding)